### DISCUSSION TITLE
Manual method to calculate the minimum_setpoint value.

### DISCUSSION THREAD

**Discussion by sergeantd83** [Alexwijn/SAT#36] [General]
1. Set Maximum Setpoint to 45 ( SAT → Configure → General → Maximum Setpoint )
2. Set your thermostat setpoint to 25 or to a setpoint that is unreachable.
3. Monitor your relative modulation level until stabilizes to a specific value and record this value.
**Note: Your relative modulation must be above 0%. If relative modulation is still 0% use higher value in step 1.**
4. Use this formula: `(( 100 - recorded relative modulation ( step 3 ) ) / 100) * 45 ( step 1 ) )` and record the result.
5. Delete your current SAT instance and re-add SAT. Then in the Calibrate System card choose the option Manually enter the overshoot protection value and enter the step 4 recorded value.

_Your system probably will overshoot during this manual calibration procedure but it is crucial for the correct SAT operation._

**Comment by dexterionut** [comment]
My room temperature is already close to 25°C (I usually keep my thermostat at 24.5°C). What value should I be using for step 2 and how does this affect the formula? - there is no mention about step 2 in it.

**Reply by dexterionut** [reply]
> My room temperature is already close to 25°C (I usually keep my thermostat at 24.5°C). What value should I be using for step 2 and how does this affect the formula? - there is no mention about step 2 in it.

Actually, I figured this out. The target temperature at step 2 is irrelevant. Step 2+3 should be read as: run you boiler as long as necessary until it reaches stable relative modulation level, above 0%. `run you boiler as long as necessary` = for example, set your thermostat to a target temperature that is unreachable. Personally, I didn't use the target temperature trick, I just deleted my SAT thermostat and manually turned on the boiler, and kept it on until my modulation level was stable and was not changing anymore. For me, that was ~15 minutes.

Also, the formula doesn't seem to be completely accurate. After I used this method and calculated a value (I got 58.2), the boiler was not running at 0% modulation with that value. I then began to test with a few temperatures like 58, 57, 56 as setpoint, for 15 minutes intervals (the amount of time took me to find a stable relative modulation), until I found a temperature that my boiler was able to hold with `0%` modulation and that was `56`.

**Reply by sergeantd83** [reply]
> set your thermostat to a target temperature that is unreachable.

I included that to the manual method, thanks for the recommendation.

> Also, the formula doesn't seem to be completely accurate. After I used this method and calculated a value (I got 58.2), the boiler was not running at 0% modulation with that value. I then began to test with a few temperatures like 58, 57, 56 as setpoint, for 15 minutes intervals (the amount of time took me to find a stable relative modulation), until I found a temperature that my boiler was able to hold with 0% modulation and that was 56.

Probably when you reached the value 58.2 the relative modulation was around 4%. If you use the formula in step 4, the minimum setpoint value is
`minimum_setpoint = ( 1 - 0.04 ) * 58.2`  which ends up as `minimum_setpoint = 55.9` is it correct?

**Reply by dexterionut** [reply]
> Probably when you reached the value 58.2 the relative modulation was around 4%. If you use the formula in step 4, the minimum setpoint value is
> `minimum_setpoint = ( 1 - 0.04 ) * 58.2` which ends up as `minimum_setpoint = 55.9` is it correct?

I used `max setpoint=60` which had a relative modulation of `3%` giving the `58.2` value using the formula. I then started testing with `max setpoint=58.2` and I was getting around 3-4% relative modulation, yes.

I'm not sure what it means tho? SAT was not using the 55.9 as a setpoint that works at 0 modulation, but rather it was always using the `58.2` calculated temp (when the automated calculation run, it gave me `58.28`)

**Reply by sergeantd83** [reply]
We need to fix the calculation mechanism, it has a bug and we are trying to find what is happening. Probably the `wait` times aren’t working properly.

**Reply by sander593** [reply]
Should this manual calculation be performed with boiler max capacity at 100% or capped at your desired level? Normally I use the boiler at minimum capacity (25%) so it runs very low.

**Reply by sergeantd83** [reply]
Is your boiler OpenTherm or EMS capable?

**Reply by sander593** [reply]
Yes it is opentherm capable. It's a Intergas Xtreme 36 connected thru a OTGW.

**Reply by sergeantd83** [reply]
Then the minimum **relative** modulation should be 0%. This is how the OpenTherm protocol implements the **relative** modulation.

**Reply by sergeantd83** [reply]
Maybe you are referring to **maximum relative modulation**. You set the max relative modulation through the MM command to 0, correct?

Then you will need to wait until you have stable boiler water temperature. This will be your Overshoot protection value.

Moreover, when you finish the initial configuration go to configure -> Advanced Settings -> Maximum Relative Modulation and set it to 0. Then your boiler will run at its lower capacity always.

**Reply by sander593** [reply]
I think we mean something different. The MM=0 command doesn't work on my boiler.
It's possible to limit the power output of the boiler to 25%. Then the boiler will never exceed a certain power output.
If I limit the max power output of the boiler to 25% it will almost never reach 45 degrees. I normally use a max temp of 35 degrees so it is not a problem.
If I set te max power output of the boiler to 100% it will reach 45 degrees pretty fast and the start to modulate.
In the first situation the result of the above calculation is (100-99,999)/100 * 45=0,0045 , which I'm unable to choose. The minimal selectable value is 10 degrees.
In the second situation the result of the above calculation is (100-64)/100 * 45=28,8.

**Reply by sergeantd83** [reply]
> In the first situation the result of the above calculation is (100-99,999)/100 * 45=0,0045 , which I'm unable to choose

If you let it rest at 45 °C then the relative modulation will be way lower than 100%, since your boiler will modulate itself and will try to modulate its power lower in order to match the boiler flow water temperature with the CS value. 

Moreover we are experimenting with a new feature for Ideal and Intergas boilers ( both of them are not respecting the MM=0 command ) and you might give it a go. Enable the `Dynamic Minimum Setpoint (Experimental)` in advanced settings and in the general settings choose the `Boiler Temperature` option of the `Minimum Setpoint Controller Version` list.

**Reply by sander593** [reply]
Thanks, I enabled your suggestions in the settings. However, the setpoint calculated by SAT is constantly lower (around 28 degrees) then the overshoot protection value (30 degrees). So the boiler cycles. Is this common behaviour of SAT? I expect a calculated setpoint above 30 degrees or is that a misinterpretation?

**Reply by sergeantd83** [reply]
Correct. It should be higher. Try the value 45 and if it still cycles raise this value further

**Reply by sander593** [reply]
I have adjusted the value and it is 50 right now. I have to further lower and see when cycling starts. I do have another question:
The setpoint remains 2 degrees below the actual temp reached by the boiler the entire time. Is this on purpose, or can you help me choose which parameter to tweak?
![image](https://github.com/user-attachments/assets/9688202e-bd8f-4384-97f6-5be4a26a430c)

**Reply by sergeantd83** [reply]
Yes, it is on purpose. Since Intergas do not respect the MM=0 command we found a workaround to suppress the modulation at its lowest levels. We do it by sending a CS that is 2 °C lower than the boiler flow water temperature. This ensures that the boiler will always work at its lowest modulation and the boiler will run as smooth as it can.

**Reply by sander593** [reply]
At night the thermostat performs perfect! But at day while the house is already heated by the sunlight, it keeps putting a little bit of heat in. In the screenshot below you can see that the boiler puts in heat while the house is already way above the setpoint. Setpoint of the house is 20 degrees, day and night.

![image](https://github.com/user-attachments/assets/2b4fd89d-0d8f-49c2-b80c-5e7ee8b14005)

![image](https://github.com/user-attachments/assets/596cd5e8-0d20-44d7-9b66-03b805a39ef8)

**Reply by sergeantd83** [reply]
You need to lower the heating curve coefficient. Even at night SAT overshoots. Try 0.2 lower coefficient than the present setting.

**Reply by sergeantd83** [reply]
Which value did you set as Overshoot protection value during initial configuration?

**Reply by sander593** [reply]
Alright, I lowered it 0,2 and will monitor it for a day.
The OPV is at 50 but it could be lower

**Reply by sergeantd83** [reply]
After examining again the first graph, I found a bug that we have when the CS is too low. That explains the reported behavior. When we will solve it, I will let you know!

**Reply by sander593** [reply]
Awesome, thank you!

**Comment by fma965** [comment]
How should TRV's be set during this manual configuration?

**Reply by sergeantd83** [reply]
Fully open. You need to set them in a high setpoint.

**Reply by fma965** [reply]
But wouldn't that mean when many are closed it won't be able to get rid of the excess heat? meaning it won't be able to run at the min setpoint?
